src/main/java/duke/Storage.java

```diff
@@ -0,0 +1,106 @@
+package duke;
+
+import duke.task.*;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.time.LocalDate;
+import java.util.ArrayList;
+import java.util.Scanner;
+
+/**
+ * A class for storing and retrieving save data for Duke.
+ */
+public class Storage {
+    File file;
+
+    /**
+     * Constructs a Storage.
+     *
+     * @param filePath String filepath of the save data.
+     */
+    public Storage(String filePath) {
+        file = new File(filePath);
+    }
+
+    /**
+     * Loads the tasks from the save data.
+     *
+     * @return ArrayList of tasks retrieved from the save data.
+     */
+    public ArrayList<Task> load() {
+        ArrayList<Task> res = new ArrayList<>();
+        try {
+            Scanner sc = new Scanner(file);
+            while (sc.hasNext()) {
+                res.add(fileLineToTask(sc.nextLine()));
+            }
+            sc.close();
+        } catch (FileNotFoundException e) {
+            try {
+                Files.createDirectories(Paths.get("./data"));
+                File f = new File("data/tasks.txt");
+            } catch (IOException ex) {
+                System.out.println("Something went wrong: " + e.getMessage());
+            }
+        }
+        return res;
+    }
+
+    /**
+     * Refreshes the save data.
+     *
+     * @param tasks TaskList to obtain new data from.
+     */
+    public void refresh(TaskList tasks) {
+        File temp = new File("data/temp.txt");
+        try {
+            FileWriter fw = new FileWriter(temp);
+            fw.write(tasks.toFileFormat());
+            fw.close();
+        } catch (IOException e) {
+            throw new DukeException(e.getMessage());
+        }
+        file.delete();
+        temp.renameTo(new File("data/tasks.txt"));
+        file = temp;
+    }
+
+    /**
+     * Adds a new task to the save data.
```

Comment:
I like that there are javadocs for every methods and constructors in all classes. Really commendable